src/main.rs

```diff
@@ -235,10 +235,25 @@ impl Args {
     /// Parse benchmarks from a buffered reader.
     fn parse_buffer<B: BufRead>(buffer: B) -> Result<Vec<Benchmark>> {
         let iter = buffer.lines();
-        let mut vec = Vec::with_capacity(iter.size_hint().0);
+        let mut vec: Vec<Benchmark> = Vec::with_capacity(iter.size_hint().0);
+        let mut ordered = true;
         for result in iter {
-            if let Ok(bench) = try!(result).parse() {
-                vec.push(bench)
+            if let Ok(bench) = try!(result).parse::<Benchmark>() {
+                if let Some(last) = vec.last() {
+                    ordered = ordered && last.name < bench.name;
+                }
+                if ordered {
+                    vec.push(bench)
```

Comment:
Is this an optimization? Is it necessary? It seems like it is uncommon and also makes this code more complex than it needs to be. If there's a really good reason for it, could you add a comment explaining why? Otherwise, I think it would be better to remove it and just use binary search and `insert`.

(The name `ordered` is also confusing because things are still ordered even when using binary search.)

Reply:
I'm not exactly sure what you mean with "this", but yes, when I saw the vector being allocated with size, I thought I'd better try to make this work at minimal cost for the existing cases. If you'd rather see the binary search unconditionally, I'm happy to oblige. I'd also try to get rid of the warnings format and clippy everything, if I were king.

Reply:
Sure, yeah, might as well get rid of the `with_capacity` thing too. I don't know why that's there.

I'd say leave the warnings and what not alone for not. And definitely don't run clippy, as I disagree with a number of its suggestions. I might get around to fixing the warnings and formatting some day, but to be honest, I don't use this tool any more and it receives just enough attention from me to keep it running.

Reply:
Oh, well, it's experienced Rusticians that directed me towards benchcmp just a few months ago. I'll be trying Criterion soon.

Now you mention it (because when I mentioned it, it didn't seep through), how can the size hint of a (buffered) file reader know how many lines are coming? Sure enough, testing with `assert_eq!(iter.size_hint().0, 0);` says it doesn't.

I've left my mark on more files this time.